lib/python/picongpu/plugins/data/compare.py

```diff
@@ -0,0 +1,222 @@
+                      species_filter="all"):
+        """
+        Return the path to the underlying data file.
+
+        Parameters
+        ----------
+        species : string
+            short name of the particle species, e.g. 'e' for electrons
+            (defined in ``speciesDefinition.param``)
+        species_filter: string
+            name of the particle species filter, default is 'all'
+            (defined in ``particleFilters.param``)
+
+        Returns
+        -------
+        A string with a path.
+        """
+        self.path_to_scan = os.path.split(os.path.split(
+                                          self.run_directory)[0])[0]
+        self.path_to_json = os.path.join(self.path_to_scan, self.json_suffix)
+        if not os.path.isfile(self.path_to_json):
+            raise IOError('The file {} does not exist.\n'
+                          'Did the simulation already run?'
+                          .format(self.path_to_json))
+        # take first value of other variable parameters
+        with open(self.path_to_json) as f:
+            range_dict = json.load(f)
+            filtered_range_dict = dict()
+            self.fix_param_dict = dict()
+            for name, attrs in range_dict.items():
+                filtered_range_dict[name] = attrs['values']
+                if ((name != variation) &
+                    (np.size(filtered_range_dict[name]) > 1)):
+                    self.fix_param_dict[name] = filtered_range_dict[name]
+        # list relevant sims in scan
+        sims_rundir = []
+        for dir_name in os.listdir(self.path_to_scan):
+            sim_path = os.path.join(self.path_to_scan, dir_name, "run")
+            if os.path.isdir(sim_path):
+                sims_rundir.append(sim_path)
+        # get species, filter
+        if species is None:
+            raise ValueError('The species parameter can not be None!')
+        if species_filter is None:
+            raise ValueError('The species_filter parameter can not be None!')
+        if variation is None:
+            raise ValueError('The variation parameter can not be None!')
+        if observable is None:
+            raise ValueError('The observable parameter can not be None!')
+        return sims_rundir, filtered_range_dict[variation]
+
+    def get_iterations(self, species, variation, observable,
+                       species_filter="all"):
+        """
+        Return an array of iterations with available data.
+
+        Parameters
+        ----------
+        species : string
+            short name of the particle species, e.g. 'e' for electrons
+            (defined in ``speciesDefinition.param``)
+        species_filter: string
+            name of the particle species filter, default is 'all'
+            (defined in ``particleFilters.param``)
+
+        Returns
+        -------
+        An array with unsigned integers.
+        """
+        return EnergyHistogramData(
```

Comment:
Is this intended to always use the `EnergyHistogramData` reader here to access iterations? Couldn't this fail if data is available for some plugins but not for others? You also seem not to need the `variation` and `observable` parameters